### PR TITLE
[ROCm] Adding a `test.is_built_with_rocm` python utility routine

### PR DESCRIPTION
**Pull request by deven-amd**
This commit adds a `test.is_built_with_rocm` python utility routine, which will return true/false based on whether TF was built with ROCm support enables (i.e. with `--config=rocm` )

This routine will be used in subsequent PRs to make ROCm specific changes to certain unit tests. The nature of these changes will be to primarily disable certain subtests, which test features that are not yet supported on the ROCm platform (for e.g. complex datatypes, 3D pooling, etc)

A `test.is_built_with_gpu_support` routine is also being added, which is essentially "is_built_with_cuda() or is_built_with_rocm()".

The other changes in this PR are ROCm implementation specific.

-------------------------------------------

@tatianashp @whchung @parallelo

### CODE REVIEW

**Review by whchung — DISMISSED** [review]

File: tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh [hunk 1]
```diff
@@ -19,21 +19,27 @@ set -e
 set -x
 
 N_JOBS=$(grep -c ^processor /proc/cpuinfo)
+N_GPUS=$(lspci|grep 'VGA'|grep 'AMD/ATI'|wc -l)
 
 echo ""
 echo "Bazel will use ${N_JOBS} concurrent job(s)."
```
Comment (chsigg) [inline tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh]: Did you want to remove this line?

**Review by chsigg — DISMISSED** [review]

File: tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh (hunk 1 above)
Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh]: yes....thank you for catching it.

let me fix that and resubmit.

Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh]: done

**Review by whchung — DISMISSED** [review]

**Review by gunan — DISMISSED** [review]

**Comment by deven-amd** [comment]
@chsigg gentle ping...this PR seems to have gotten stuck in merge pipe.

File: tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh [hunk 2]
```diff
@@ -19,21 +19,26 @@ set -e
...
 export CC_OPT_FLAGS='-mavx'
 
 export TF_NEED_ROCM=1
+export TF_GPU_COUNT=${N_GPUS}
 
 yes "" | $PYTHON_BIN_PATH configure.py
 
 # Run bazel test command. Double test timeouts to avoid flakes.
-bazel test --config=rocm --test_tag_filters=-no_oss,-oss_serial,-no_gpu,-benchmark-test -k \
+bazel test --config=rocm --test_tag_filters=-no_oss,-oss_serial,-no_gpu,,-no_rocm,-benchmark-test -k \
```
Comment (chsigg) [inline tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh]: Remove extra comma.

**Review by chsigg — DISMISSED** [review]

File: tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh (hunk 2 above)
Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cc_core.sh]: done

**Review by whchung — DISMISSED** [review]

**Review by chsigg — DISMISSED** [review]

**Comment by deven-amd** [comment]
how do I go about identifying and fixing the "py_lint" errors? 
Is there a `clang-format` equivalent I should be running on the python source code changes?
thanks

**Comment by rthadur** [comment]
@deven-amd please fix the lint error below : 
`FAIL: Found 1 non-whitelited pylint errors:
tensorflow/python/framework/test_util.py:1382: [C0301(line-too-long), ] Line too long (81/80)`

**Comment by deven-amd** [comment]
> @deven-amd please fix the lint error below :
> `FAIL: Found 1 non-whitelited pylint errors: tensorflow/python/framework/test_util.py:1382: [C0301(line-too-long), ] Line too long (81/80)`

@rthadur done.

I also fixed the failure in the `//tensorflow/python:build_info_test ` test. That required adding some more code, and re-arranging the PR into different commits.

@chsigg , please re-review. thanks

**Review by whchung — DISMISSED** [review]

**Review by chsigg — DISMISSED** [review]

**Comment by deven-amd** [comment]
@rthadur @chsigg 

pushed out two more changes
1. updated the `gen_build_info.py` file to remove the lint errors (ran it through `yapf`, hopefully it fixed all lint issues)
2. updated test goldens for the `//tensorflow/tools/api/tests:api_compatibility_test`. This is to account for the addition of the two new routines `is_built_with_rocm` and `is_built_with_gpu_support` to the API

please re-review and approve.

thanks

**Review by whchung — DISMISSED** [review]

**Comment by deven-amd** [comment]
@rthadur 

running `yapf` seems to have made things worse from the linter's point of view.

How do I go about running the `do_pylint` check locally, so that I do not run into failures at CI check time? 

This PR is a blocker for several others. Those other PRs also involve updating python files, and my updates to them will run the risk of failing the `do_pylint` step too. Knowing how to run that step locally will be huge help. 

Thanks

deven

**Comment by rthadur** [comment]
@deven-amd you can check the sanity logs here https://source.cloud.google.com/results/invocations/29049114-e768-46e8-8026-ae3e554a516c/log 
and alternatively you can run this command  locally `tensorflow/tools/ci_build/ci_sanity.sh` to run all the checks in the build.

**Comment by deven-amd** [comment]
@rthadur thank you for instructions on how to run the link check locally.
I have pushed out another commit with a reformatted version of `gen_build_info.py`...it now passes the lint check for me locally...keeping my fingers crossed that the same holds true for the actual CI run as well.

@chsigg , please re-re-review and re-re-re-bless it :)

thanks again

**Review by whchung — APPROVED** [review]

**Comment by deven-amd** [comment]
any update on this PR?

**Comment by chsigg** [comment]
This PR includes an API change, which has a separate process that I'm not familiar with. I've asked internally what the next steps are. Thank you for the patience.

**Comment by deven-amd** [comment]
@chsigg thank you for the update. 

As usual, we really appreciate all the work you have (and continue to) put in, in getting our PRs merged. 

thanks again

deven

**Comment by jhseu** [comment]
Removing the API review tag since it was reviewed internally and the API change was approved.

**Comment by rthadur** [comment]
@chsigg can you please approve the PR

**Comment by chsigg** [comment]
This PR has been merged in commit d0814e1.

**Pull request closed without merging**